apps/wallet/src/ui/app/hooks/useDeepBook.ts

```diff
@@ -145,28 +156,242 @@ export function useBalanceConversion(
+	const activeAccountAddress = activeAccount?.address;
+	const signer = useSigner(activeAccount);
+
+	return useMutation({
+		mutationKey: [DEEPBOOK_KEY, 'create-account', activeAccountAddress],
+		mutationFn: async () => {
+			if (activeAccountAddress) {
+				const txb = deepBookClient.createAccount(activeAccountAddress);
+
+				return signer?.signAndExecuteTransactionBlock({ transactionBlock: txb });
+			}
+
+			return null;
+		},
+		onSuccess,
+	});
+}
+
+export function useMarketAccountCap(activeAccountAddress?: string) {
+	const {
+		data,
+		isLoading: getOwnedObjectsLoading,
+		isError: getOwnedObjectsError,
+		isRefetching: isRefetchingOwnedObjects,
+		refetch,
+	} = useGetOwnedObjects(
+		activeAccountAddress,
+		{
+			MatchAll: [{ StructType: '0xdee9::custodian_v2::AccountCap' }],
+		},
+		10,
+	);
+
+	const objectId = data?.pages?.[0]?.data?.[0]?.data?.objectId;
+
+	const {
+		data: suiObjectResponseData,
+		isLoading: getObjectLoading,
+		isRefetching: isRefetchingObject,
+		isError: getObjectError,
+	} = useGetObject(objectId);
+
+	const fieldsData =
+		suiObjectResponseData?.data?.content?.dataType === 'moveObject'
+			? (suiObjectResponseData?.data?.content?.fields as Record<string, string | number | object>)
+			: null;
+
+	return {
+		data: fieldsData,
+		isLoading: getObjectLoading || getOwnedObjectsLoading,
+		isError: getOwnedObjectsError || getObjectError,
+		isRefetching: isRefetchingObject || isRefetchingOwnedObjects,
+		refetch,
+	};
+}
+
+const MAX_COINS_PER_REQUEST = 10;
+
+export async function getCoinsByBalance({
+	coinType,
+	balance,
+	suiClient,
+	address,
+}: {
+	coinType: string;
+	balance: string;
+	suiClient: SuiClient;
+	address: string;
+}) {
+	let cursor: string | undefined | null = null;
+	let currentBalance = 0n;
+	let hasNextPage = true;
+	const coins = [];
+	const bigIntBalance = BigInt(balance);
+
+	outer: while (currentBalance < bigIntBalance && hasNextPage) {
+		const { data, nextCursor } = await suiClient.getCoins({
+			owner: address,
+			coinType,
+			cursor,
+			limit: MAX_COINS_PER_REQUEST,
+		});
+
+		if (!data || !data.length) {
+			break;
+		}
+
+		for (const coin of data) {
+			currentBalance += BigInt(coin.balance);
+			coins.push(coin);
+
+			if (currentBalance >= bigIntBalance) {
+				break outer;
+			}
+		}
+
+		cursor = nextCursor;
+		hasNextPage = !!nextCursor;
+	}
+
+	return coins;
+}
+
+export async function getPlaceMarketOrderTxn({
+	deepBookClient,
+	poolId,
+	balanceInMist,
+	accountCapId,
+	coins,
+	coinType,
+}: {
+	deepBookClient: DeepBookClient;
+	poolId: string;
+	balanceInMist: string;
+	accountCapId: string;
+	coins: CoinStruct[];
+	coinType: string;
+}) {
+	const txb = new TransactionBlock();
+
+	let baseCoin;
+	if (coinType === SUI_TYPE_ARG) {
+		baseCoin = txb.splitCoins(txb.gas, [balanceInMist]);
+	} else {
+		const primaryCoinInput = txb.object(coins[0].coinObjectId);
+
+		txb.mergeCoins(
+			primaryCoinInput,
+			coins.map((coin) => txb.object(coin.coinObjectId)),
```

Comment:
I think you need to skip the 0-index coin here otherwise you're merging the coin into itself